File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/builder/vechicle/VehicleBuilder.java
package com.bigboxcode.designpattern.builder.vechicle;

public interface VehicleBuilder {

    void addWheel(int noOfWheel);

    void addEngine(int noOfEngine);

    void addSeat(int noOfSeat);

    void addInterior();

    void addDoor(int noOfDoor);

    void addWing(int noOfWing) throws Exception;



}

File: design-patterns/php/builder/vehicle/VehicleBuilder.php
<?php
// VehicleBuilder.php

namespace BigBoxCode\DesignPattern\Builder\Vehicle;

interface VehicleBuilder {
    public function addWheel(int $noOfWheel): void;
    public function addEngine(int $noOfEngine): void;
    public function addSeat(int $noOfSeat): void;
    public function addInterior(): void;
    public function addDoor(int $noOfDoor): void;
    public function addWing(int $noOfWing): void;
}